### PR TITLE
Use clang-format to format the entire code-base

### PR DESCRIPTION
**Pull request by kylef**
This PR was from @tjanc suggested clang-format.

@klokane @pksunkara Do you have any major objections?

### CODE REVIEW

File: src/Blueprint.cc [hunk 1]
```diff
@@ -10,81 +10,64 @@
...
 
-Element::Element(const Element& rhs)
-{
-    this->element = rhs.element;
+Element::Element(const Element &rhs) {
+    this->element    = rhs.element;
     this->attributes = rhs.attributes;
-    this->content = rhs.content;
-    this->category = rhs.category;
+    this->content    = rhs.content;
```
Comment (kylef) [inline src/Blueprint.cc]: I do not like `AlignConsecutiveAssignments` because if there are any changes to a set of variables it is likely that clang-format will reformat other variables. This will create larger diffs with unrelated formatting when anything is changed in the assignments list.

File: src/AssetParser.h [hunk 2]
```diff
@@ -15,109 +15,110 @@
...
+const char *const SchemaRegex = "^[[:blank:]]*[Ss]chema[[:blank:]]*$";
+
+/**
+ *  Asset Section Processor
+ */
+template <> struct SectionProcessor<Asset> : public SectionProcessorBase<Asset>
+{
+
+    static MarkdownNodeIterator
+    processSignature(const MarkdownNodeIterator &node,
```
Comment (klokane) [inline src/AssetParser.h]: do not like this break (return type vs function name on difrerent line)

File: src/Blueprint.h [hunk 3]
```diff
@@ -25,465 +25,484 @@
...
+    /** Enumeration of possible values */
+    Values values;
+};
 
-    /** Parameter with MSON-like syntax */
-    struct MSONParameter : public Parameter {
-    };
+/** Parameter with MSON-like syntax */
+struct MSONParameter : public Parameter
+{};
```
Comment (klokane) [inline src/Blueprint.h]: This is strange. Compare with
https://github.com/apiaryio/snowcrash/pull/434/files#diff-2c45c82fe0dd50c00a34ebdfb3dec1f9R70

```
SourceMap<Element>::~SourceMap() {}
```

Why are in this case empty brackets on separated line, while second case it is on same line

**Comment by klokane** [comment]
BTW no `ColumnLimit` is specified. Default value is 80?
I would like allow 120. There is no reason today be on smaller value.

File: src/MSONOneOfParser.cc [hunk 4]
```diff
@@ -12,121 +12,119 @@
 
 namespace snowcrash {
 
-    /** Implementation of processNestedSection */
-    MarkdownNodeIterator SectionProcessor<mson::OneOf>::processNestedSection(const MarkdownNodeIterator& node,
-                                                                             const MarkdownNodes& siblings,
-                                                                             SectionParserData& pd,
-                                                                             const ParseResultRef<mson::OneOf>& out) {
+/** Implementation of processNestedSection */
+MarkdownNodeIterator SectionProcessor<mson::OneOf>::processNestedSection(
```
Comment (klokane) [inline src/MSONOneOfParser.cc]: There are parameters breaked on "indentation" while in many other cases is break on "parenthesis"
```
MarkdownNodeIterator
SectionProcessor<mson::OneOf>::processNestedSection(const MarkdownNodeIterator &node, 
                                                                                             const MarkdownNodes &siblings,
```

**Comment by tjanc** [comment]
[Use clang-format to format the whole code base](https://trello.com/c/wY5thN9K/780-use-clang-format-to-format-the-whole-code-base)

**Comment by tjanc** [comment]
@klokane Yes, the column limit used by LLVM is 80. Although historically the reason for an 80 column limit is monitor size, many code bases still use it for different reasons:

- Typography 101: shorter lines are easier to read
- Two or three vertical buffers are feasible; especially useful for us vim users, but also for anyone doing a diff or 3-way merge.
- Embedding code into documents, presentations or blog posts as-is
- One is forced to be concise about naming things (proper use of namespaces etc)

Do you really think a column limit of 120 beats 80 in terms of readability?
I think clang-format does a good job at splitting lines.

File: src/Blueprint.h (hunk 3 above)
Comment (tjanc) [inline src/Blueprint.h]: Because the first is a struct, whereas the second is a method.

File: src/AssetParser.h (hunk 2 above)
Comment (tjanc) [inline src/AssetParser.h]: I'm not a fan either, thanks.

File: src/Blueprint.cc (hunk 1 above)
Comment (tjanc) [inline src/Blueprint.cc]: That's a good catch; thanks.

**Comment by klokane** [comment]
In general I've no problem with 80 itselt. But I do not like for this case indentation on parenthesis.

```
MarkdownNodeIterator
SectionProcessor<mson::OneOf>::processNestedSection(const MarkdownNodeIterator &node,
                                                    const MarkdownNodes &siblings,
```

Instead of i mean much better readability is indented on "tab"
```
MarkdownNodeIterator
SectionProcessor<mson::OneOf>::processNestedSection(
    const MarkdownNodeIterator &node,
    const MarkdownNodes &siblings,
   ....) {
...
}
```

**Comment by tjanc** [comment]
@klokane I will modify the config to come as close to this suggestion as possible.

**Comment by tjanc** [comment]
@klokane @kylef I've modified the clang-format according to your suggestions. How do you like the formatting now?

File: src/ActionParser.h [hunk 5]
```diff
@@ -20,543 +20,603 @@
...
+
+    static MarkdownNodeIterator processSignature(
+        const MarkdownNodeIterator &node,
+        const MarkdownNodes &siblings,
+        SectionParserData &pd,
+        SectionLayout &layout,
+        const ParseResultRef<Action> &out) {
+
+        actionHTTPMethodAndName(
+            node, out.node.method, out.node.name, out.node.uriTemplate);
```
Comment (klokane) [inline src/ActionParser.h]: This is strange, it is possible to break line before param which exceed column limit?

**Comment by klokane** [comment]
With other I'm fine

**Comment by pksunkara** [comment]
Let's keep `NamespaceIndentation: All`. We kinda started doing that and now changing it means we lose a lot of git blame history since this format commit changes almost every line in the code.

**Comment by pksunkara** [comment]
And what's the reason for going LLVM vs Webkit? And to be honest, I like #380 more.

**Comment by pksunkara** [comment]
Closing this in favor of Thomas's new PR regarding the 80 column limit.
